onnxruntime/test/common/tensor_op_test_utils.h

```diff
@@ -20,9 +20,21 @@ namespace onnxruntime {
 namespace test {
 
 namespace detail {
-inline int64_t SizeFromDims(gsl::span<const int64_t> dims) {
-  const int64_t size = std::accumulate(
-      dims.cbegin(), dims.cend(), static_cast<int64_t>(1), std::multiplies<int64_t>{});
+inline int64_t SizeFromDims(gsl::span<const int64_t> dims, gsl::span<const int64_t> strides = {}) {
+  int64_t size = 1;
+  if (strides.empty()) {
+    size = std::accumulate(dims.cbegin(), dims.cend(), static_cast<int64_t>(1), std::multiplies<int64_t>{});
+  } else {
+    ORT_ENFORCE(dims.size() == strides.size());
+    for (size_t dim = 0; dim < dims.size(); ++dim) {
+      if (dims[dim] == 0) {
```

Comment:
in which case we have such a dims?

Reply:
Shape[2,3,3,0,3] is a valid shape, right?